### PR TITLE
add progressively function and remove progress from from_url

### PR DESCRIPTION
**Pull request by tanho63**
`progressively` now handles the progress signalling instead of `from_url`. Will need an accompanying vignette update, probably. 

This now improves the functions that use a `dplyr::filter` for seasons, and doesn't hurt the `map_dfr` functions at all. 

Slightly more awkward to use for package devs, perhaps, but it's now a lot like the way purrr's quietly, possibly, safely work so I'm happy with it

### CODE REVIEW

**Comment by mrcaseb** [comment]
I think I like the idea but can't play around with it because I am on mobile. 

One thing: what if p is not NULL but also not a progressor?

The old code checked that with `inherits`. Would the function break?

**Comment by tanho63** [comment]
could add it to progressively, i'll look

File: tests/testthat/test-nflverse.R [hunk 1]
```diff
@@ -105,11 +105,13 @@ test_that("Cache clearing works",{
   skip_on_cran()
   skip_if_offline("github.com")
 
-  expect(memoise::has_cache(load_player_stats)(), "Function was not memoised")
+  rds_from_url("https://raw.githubusercontent.com/nflverse/nfldata/master/data/draft_picks.rds")
```
Comment (mrcaseb) [inline tests/testthat/test-nflverse.R:108]: What is this line doing in the test? The data isn't assigned or within an expect function.

File: R/from_url.R [hunk 2]
```diff
@@ -100,3 +83,26 @@ qs_from_url <- function(url, ...){
 
   return(content)
 }
+
+#' Progressively
+#'
```
Comment (mrcaseb) [inline R/from_url.R:88]: Can you add a description here that explains the function a bit?

**Review by mrcaseb — APPROVED** [review]
Just two minor things but the rest looks good to me. Interesting approach!

File: tests/testthat/test-nflverse.R (hunk 1 above)
Comment (tanho63) [reply tests/testthat/test-nflverse.R:108]: This is testing that the function is cached, so first we need to make a successful function call, then we need to test that it remembers that argument was already executed once (the has_cache() line), then we clear it (and expect it to run successfully + print the correct message), then we test to see if the function has a cache for the argument (it now should not).

File: R/from_url.R (hunk 2 above)
Comment (tanho63) [reply R/from_url.R:88]: Added.

**Comment by mrcaseb** [comment]
I think this is good now. Not merging in case you want to add something

**Pull request merged**